### PR TITLE
protocol/validation: move consensus program exec into a new function

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

**Comment by kr** [comment]
Not ready for review yet; the callers still need to be updated. But you can get the idea.

**Comment by kr** [comment]
PTAL.

It is a little scary that none of the unit tests broke when I changed the signature of a function called ValidateBlock in package chain/protocol/validation.

**Comment by bobg** [comment]
...which sounds like there oughta be a couple of unit tests in this PR.

**Comment by kr** [comment]
Maybe in a separate PR, yeah.

File: protocol/validation/validation.go [hunk 1]
```diff
@@ -376,15 +376,21 @@ func checkValidDest(vs *validationState, vd *bc.ValueDestination) error {
 	return nil
 }
 
+// ValidateBlockSig runs the consensus program from prev on b.
+func ValidateBlockSig(b, prev *bc.BlockEntries) error {
+	vmContext := NewBlockVMContext(b, prev.Body.NextConsensusProgram, b.Witness.Arguments)
```
Comment (bobg) [inline protocol/validation/validation.go]: Suggest some bulletproofing here against the possibility that `prev == nil`.

Comment (kr) [inline protocol/validation/validation.go]: I will update the godoc to specify that both b and prev must be non-nil.

Comment (bobg) [inline protocol/validation/validation.go]: Y'know, haha, no bulletproofing would be needed if we did the thing here that we were talking about in [in this thread](https://github.com/chain/chain/pull/788#discussion_r108822154): have callers pass not `prev` but only its `NextConsensusProgram`, which is all this function wants.

**Comment by kr** [comment]
Rebased. Need to update for the new unit tests.

File: protocol/validation/validation.go (hunk 1 above)
Comment (kr) [inline protocol/validation/validation.go]: 😮  Will do!

**Comment by kr** [comment]
PTAL

**Comment by bobg** [comment]
LGTM

**Pull request merged**